lib/chef/provider/remote_directory.rb

```diff
@@ -67,13 +67,10 @@ def ls(path)
                  ::File::FNM_DOTMATCH)
 
         # Remove current directory and previous directory
-        files.reject! do |name|
+        files = files.reject do |name|
```

Comment:
Oh yes, good point. `files = ` done before the map step was removed that used it. I'll update the pr